testdata/p4_16_samples/array1.p4

```diff
@@ -0,0 +1,71 @@
+#include <core.p4>
+#include <v1model.p4>
+
+header data_t {
+    bit<32>     f1;
+    bit<32>     f2;
+    bit<16>     h1;
+    bit<16>     h2;
+    bit<8>[16]  arr;
+}
```

Comment:
Could you please add positive and/or error test cases similar to the following to demonstrate what is expected when the array type is an array type or contains array types?
```
struct S {
    bit<16> f1;
    bit<8>[16] nested_arr;
    bit<16> f2;
}

header data_t {
    bit<32>     f1;
    bit<32>     f2;
    bit<16>     h1;
    bit<16>     h2;
    S[16]  arr;
}
...
    hdr.data.arr[x].nested_arr[y] = ...;
    ... = hdr.data.arr[z].nested_arr[w];
...
```
and (if the grammar allows it (I did not look very closely at changes to `p4parser.ypp`)):
```
header data_t {
    bit<32>     f1;
    bit<32>     f2;
    bit<16>     h1;
    bit<16>     h2;
    bit<8>[16][16]  arr;
}
...
    hdr.data.arr[x][y] = ...;
    ... = hdr.data.arr[z][w];
...
```

Reply:
While these are allowed, perhaps they shouldn't be.  They end up running extremely slowly, as the HSIndexSimplifier pass (which replaces non-constant indexes with if trees) winds up replcating this assignment 65536 times (all the possible combinations of indexes for x, y, z, and w) with as many ifs.  Not terribly practical.

Reply:
Explicitly disallowing nested arrays is completely fine with me (as long as it's made clear in the spec).